lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1651,3 +1670,301 @@ def test_bumper_metadata(self, get_url_for_profiles, get_bumper_settings, is_bum
+                'sub': 'a_sub_file.srt.sjson',
+                'sources': sources,
+                'poster': None,
+                'captionDataDir': None,
+                'showCaptions': 'true',
+                'generalSpeed': 1.0,
+                'speed': None,
+                'savedVideoPosition': 0.0,
+                'start': 3603.0,
+                'end': 3610.0,
+                'transcriptLanguage': 'en',
+                'transcriptLanguages': OrderedDict({'en': 'English', 'uk': u'Українська'}),
+                'ytTestTimeout': 1500,
+                'ytApiUrl': 'https://www.youtube.com/iframe_api',
+                'ytMetadataUrl': 'https://www.googleapis.com/youtube/v3/videos/',
+                'ytKey': None,
+                'transcriptTranslationUrl': self.item_descriptor.xmodule_runtime.handler_url(
+                    self.item_descriptor, 'transcript', 'translation/__lang__'
+                ).rstrip('/?'),
+                'transcriptAvailableTranslationsUrl': self.item_descriptor.xmodule_runtime.handler_url(
+                    self.item_descriptor, 'transcript', 'available_translations'
+                ).rstrip('/?'),
+                'autohideHtml5': False,
+                'recordedYoutubeIsAvailable': True,
+            })),
+            'track': None,
+            'transcript_download_format': u'srt',
+            'transcript_download_formats_list': [
+                {'display_name': 'SubRip (.srt) file', 'value': 'srt'},
+                {'display_name': 'Text (.txt) file', 'value': 'txt'}
+            ],
+            'poster': 'null'
+        }
+
+        with override_settings(FEATURES=self.FEATURES):
+            expected_content = self.item_descriptor.xmodule_runtime.render_template('video.html', expected_context)
+
+        self.assertEqual(content, expected_content)
+
+        # Enable flag and check that it's enabled
+        self.FEATURES.update({"ENABLE_AUTOADVANCE_VIDEOS": True})
+
+        with override_settings(FEATURES=self.FEATURES):
+            content = self.item_descriptor.render(STUDENT_VIEW).content
+        expected_context['autoadvance_enabled'] = True
+        with override_settings(FEATURES=self.FEATURES):
+            expected_content = self.item_descriptor.xmodule_runtime.render_template('video.html', expected_context)
+        self.assertEqual(content, expected_content)
+
+    def test_autoadvance_in_video_depends_from_course_advanced_settings(self):
+        """
+        Check that the course-level settings (advanced settings) impact whether
+        the current video will auto-advance or not. This test presumes
+        video-advance is globally enabled.
+        """
+        self.FEATURES.update({
+            "ENABLE_AUTOADVANCE_VIDEOS": True,
+        })
+
+        # FIXME enable at course-level, set to true. The following are attempts to do it. Delete until the -----
+
+        self.item_descriptor.video_auto_advance=114422 # True
```

Comment:
@bdero @bradenmacdonald  In this section I attempt to change the video_auto_advance course setting in many ways (and many more that I didn't include). The values I use are just to test whether it changes or not. I manage to change the value. The problem is that render() in line 1900 doesn't use my value and "content" always returns with autoAdvance:False (it should be True). I need something to update the value everywhere, but I don't know how to connect all objects I found (CourseDescriptor, VideoDescriptor, LmsFieldData, CourseFactory, store, …). I looked for similar examples for other fields from inheritance.py but noone handles the render(), and many don't have tests. I did a lot of debugging and found too much information on related things but I don't know the correct approach for this case.

Reply:
@clemente It looks like you're changing the field on the video descriptor and then calling update on the course descriptor instead of the video descriptor. here's a snippet from my local testing:
```python
>>> self.item_descriptor.video_auto_advance = True
>>> self.store.update_item(self.item_descriptor, self.user.id)
>>> with override_settings(FEATURES=self.FEATURES):
...     content = self.item_descriptor.render(STUDENT_VIEW).content
...
>>> "'autoadvance_enabled': True" in content
True
```

Note that child xblocks inherit field data from parent xblocks, so this should work equally well if you set the attribute on the course that `item_descriptor` is a part of and save it to the modulestore.

I also noticed that you tried creating a brand new course before calling update on it. Even if the  `video_auto_advance` attribute is properly set when creating the new course, this won't work as written either because the video module being rendered is a child of the old course, not the new course.

Reply:
Thanks for explaining.
    >>> "'autoadvance_enabled': True" in content
    True
Note that the variable I'm trying to set to True is `autoAdvance` („should this video auto-advance“), not the feature flag `autoadvance_enabled` („are the controls visible“). Could you check with `autoAdvance`?

I tried with different methods and applied your comments (changing in parent, avoiding changing course), and also comments from Braden. I pushed the new code. But the variable still doesn't change. I'll try a bit more later.

Reply:
@clemente `autoAdvance` is not a python variable (as you can tell be the lower camel casing), so I'm not sure what you mean.

In the code, you set the JS context variable as:
```python
'autoAdvance': self.auto_advance or getattr(self, 'video_auto_advance', False),
```
So it seems that the variable you should be setting and reading in the tests is `video_auto_advance`.

Reply:
@bradenmacdonald exactly, that's what I'm doing: setting video_auto_advance so that it ends up in the context as autoAdvance. I set video_auto_advance, I read autoAdvance.

Reply:
I'm happy to announce that this approach (`self.item_descriptor.xmodule_runtime.xmodule_instance.video_auto_advance`) did effectively change the rendered variable: it modifies the parameter `video_auto_advance` which then is processed by `video_module.py` and then rendered as `autoAdvance`. All without using `save()`/`update_item()`.
I removed all FIXMEs and internal tests. I fixed the test logic and the test passes :-)
Thanks @bdero